drmad_mnist.lua

```diff
@@ -234,33 +239,48 @@ local dHVP = grad(gradProj)
 
 ----------------------------------------------
 -- Backpropagate the validation errors
-numIter = numEpoch * (trainData.size)
+numIter = numEpoch * (epochSize == -1 and trainset:size() or epochSize)
 local beta = th.linspace(0.001, 0.999, numIter)
 
 -- learning rate for hyperparameters
 local hLr = 0.01
 
+local buffer
 for epoch = 1, numEpoch do
 
     print('Backword Training Epoch #' .. epoch)
-    for i = 1, trainData.size do
+    for i, inputs, targets in trainset:subiter(1, epochSize) do
         -- Next sample:
-        local x = trainData.x[i]:view(1, inputSize)
-        local y = th.view(trainData.y[i], 1, 10)
+        local x, y = makesample(inputs, targets)
+
         for j = 1, nLayers do
-            params.W[j] = th.mul(initParams.W[j], (1 - beta[i + (numEpoch * (epoch-1))])) +
-                    th.mul(finalParams.W[j], beta[i + (numEpoch * (epoch-1))])
-            DV[j] = DV[j] + validGrads.W[j] * eLr
```

Comment:
@bigaidream this line was the issue. `DV1` and `DV[1]` point to different tensors after you do `DV[j] = DV[j]`